src/wait.ts

```diff
@@ -1,43 +1,61 @@
-import { Run } from "./github";
+import { Run, OctokitGitHub, GitHub } from "./github";
+import { Input, parseInput } from "./input";
 
 export interface Wait {
   wait(secondsSoFar?: number): Promise<number>;
 }
 
 export class Waiter implements Wait {
   private readonly info: (msg: string) => void;
-  private readonly getRun: () => Promise<Run>;
-  private readonly pollIntervalSeconds: number;
-  private readonly continueAfterSeconds: number | undefined;
+  private input: Input;
+  private githubClient: GitHub;
+  private workflowId: any;
+
   constructor(
-    getRun: () => Promise<Run>,
-    pollIntervalSeconds: number,
-    continueAfterSeconds: number | undefined,
+    workflowId: any,
+    githubClient: GitHub,
+    input: Input,
     info: (msg: string) => void
   ) {
-    this.getRun = getRun;
-    this.pollIntervalSeconds = pollIntervalSeconds;
-    this.continueAfterSeconds = continueAfterSeconds;
+    this.workflowId = workflowId;
+    this.input = input;
+    this.githubClient = githubClient;
     this.info = info;
   }
 
   wait = async (secondsSoFar?: number) => {
+    const runs = await this.githubClient.runs(
+      this.input.owner,
+      this.input.repo,
+      this.input.branch,
+      this.workflowId
+    );
+    const previousRun = runs
+      .filter(run => run.id < this.input.runId)
+      .sort((a, b) => b.id - a.id)[0];
+
     if (
-      this.continueAfterSeconds &&
-      (secondsSoFar || 0) >= this.continueAfterSeconds
+      this.input.continueAfterSeconds &&
+      (secondsSoFar || 0) >= this.input.continueAfterSeconds
     ) {
       this.info(`🤙Exceeded wait seconds. Continuing...`);
       return secondsSoFar || 0;
     }
-    const run = await this.getRun();
+
```

Comment:
An aha thought accurred to me when trying to understand the implications of changing the sort order above.

Previously we looked for all in progress runs filtering for the previous run then the waiter held a ref it it's I'd to poll for it's completed status.

Now that we are fetching all in progress runs inside the waiter, we technically don't need to check for completed status of any run since we only get back in progress runs. 

I think that we can get away with just checking if there are any runs in this in progess list that have a run id less than the current run id.

Make sense? I'm happy to merge and follow up unless you want to grab that simplification here.

Reply:
Yeah you are right. Good catch. I have updated the PR to reflect this change.